Prepare release 3.1.0 (#585)
Parsely/wp-parsely: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [3.1.0] 2022-01-12

The 3.1.0 release is a minor release for the plugin that does not introduce any breaking changes coming from the 3.0 branch. This version's primary focus is adding support for WordPress decoupled architectures and a revamped settings page. We have also worked hard on refining our code, testing, and delivery process.

> The Parse.ly plugin now hooks into the WordPress REST API to provide the Parse.ly metadata inserted automatically in non-decoupled sites. No new endpoint is introduced -- a `parsely` field is now rendered in the tracked objects containing that data (e.g., `post` and `page`). Please, refer to the plugin's README file for more details. Note that the tracking script must be inserted manually in the decoupled front-end. This behavior can be disabled using a filter.

Reviewer: The final sentence here reads as though the behaviour of manually inserting a tracking script can be disabled using a filter. I think the last two sentences need switching order.

- Revamped wp-admin settings page, divided in sections. [#518](https://github.com/Parsely/wp-parsely/pull/518)
- Show error in settings page when duplicated tracking is selected. [#543](https://github.com/Parsely/wp-parsely/pull/543)
- Instructions for local development. [#525](https://github.com/Parsely/wp-parsely/pull/525)
- Local developer environment logs command. [#532](https://github.com/Parsely/wp-parsely/pull/532)
- Husky-based git commit hooks to enforce linting rules prior to commit. [#538](https://github.com/Parsely/wp-parsely/pull/538)
- Linting for JavaScript and CSS files. [#527](https://github.com/Parsely/wp-parsely/pull/527)
- Types to function arguments in GetCurrentUrlTest. [#504](https://github.com/Parsely/wp-parsely/pull/504)
- End-to-end test to verify if scripts are rendered in the front-end. [#528](https://github.com/Parsely/wp-parsely/pull/528)
- Concurrency to CI configuration and Composer tweaks. [#559](https://github.com/Parsely/wp-parsely/pull/559)
- Explicit dependabot reviewers on GitHub. [#526](https://github.com/Parsely/wp-parsely/pull/526)
- WordPress.org banner images. [#581](https://github.com/Parsely/wp-parsely/pull/581)

### Changed

> - Hiding _Disable AMP_ section on settings page when the AMP plugin is not enabled. [#519](https://github.com/Parsely/wp-parsely/pull/519)

Reviewer: ```suggestion
- Hiding _Disable AMP_ field on settings page when the AMP plugin is not enabled. [#519](https://github.com/Parsely/wp-parsely/pull/519)
```

- Use built-in WordPress submit button instead of custom one in the settings page. [#513](https://github.com/Parsely/wp-parsely/pull/513)
- Improved wp-admin settings page help texts. [#552](https://github.com/Parsely/wp-parsely/pull/552)
- `@wordpress/scripts` bumped from 19.2.1 to 19.2.2. [#503](https://github.com/Parsely/wp-parsely/pull/503)
- `prettier` bumped from 2.4.1 to 2.5.0. [#509](https://github.com/Parsely/wp-parsely/pull/509) 
- `concurrently` bumped from 6.4.0 to 6.5.1. [#551](https://github.com/Parsely/wp-parsely/pull/551)

> - Ubuntu bumped in from 18.04 to 20.04 on CI tests. [#445](https://github.com/Parsely/wp-parsely/pull/445)

Reviewer: ```suggestion
- Ubuntu bumped from 18.04 to 20.04 on CI tests. [#445](https://github.com/Parsely/wp-parsely/pull/445)
```

- Unit and Integration tests run in random order. [#511](https://github.com/Parsely/wp-parsely/pull/511)
- Correct Parse.ly spelling in tests comments. [#561](https://github.com/Parsely/wp-parsely/pull/561)
- Minor amendments on the documentation. [#514](https://github.com/Parsely/wp-parsely/pull/514)
- Updated release process guidelines. [#567](https://github.com/Parsely/wp-parsely/pull/567)
- Removed checkboxes from GitHub's PR template. [#512](https://github.com/Parsely/wp-parsely/pull/512)
- Improved JS scripts integration tests. [#557](https://github.com/Parsely/wp-parsely/pull/557)
- WordPress.org screenshots for settings page. [#574](https://github.com/Parsely/wp-parsely/pull/574)

### Fixed

- Undefined index error on settings page. [#536](https://github.com/Parsely/wp-parsely/pull/536)
- Source the correct asset for the Recommendations Widget. [#545](https://github.com/Parsely/wp-parsely/pull/545)
- End-to-end tests in CI (GitHub Actions). [#521](https://github.com/Parsely/wp-parsely/pull/521)

> - Source code linting violations (PHPCS with --serverity=1). [#544](https://github.com/Parsely/wp-parsely/pull/544)

Reviewer: ```suggestion
- Source code linting violations (PHPCS with `--serverity=1`). [#544](https://github.com/Parsely/wp-parsely/pull/544)
```

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [3.1.0] 2022-01-12

The 3.1.0 release is a minor release for the plugin that does not introduce any breaking changes coming from the 3.0 branch. This version's primary focus is adding support for WordPress decoupled architectures and a revamped settings page. We have also worked hard on refining our code, testing, and delivery process.

> The Parse.ly plugin now hooks into the WordPress REST API to provide the Parse.ly metadata inserted automatically in non-decoupled sites. No new endpoint is introduced -- a `parsely` field is now rendered in the tracked objects containing that data (e.g., `post` and `page`). This behavior can be disabled using a filter. Please, refer to the plugin's README file for more details. Note that the tracking script must be inserted manually in the decoupled front-end.

Reviewer 2: ```suggestion
The Parse.ly plugin now hooks into the WordPress REST API to provide content metadata in a format that's easy for a variety of client applications to consume. A `parsely` field containing the metadata is now rendered in the tracked objects (e.g., `post` and `page`). No new endpoint is introduced. This behavior can be disabled using a filter. Please, refer to the plugin's README file for more details. Note that the tracking script must still be inserted manually in the decoupled front-end or otherwise loaded for your site.
```

Reviewer 3: Maybe it would also be worth it to mention the filter name here?

- Ubuntu bumped from 18.04 to 20.04 on CI tests. [#445](https://github.com/Parsely/wp-parsely/pull/445)
- Unit and Integration tests run in random order. [#511](https://github.com/Parsely/wp-parsely/pull/511)
- Correct Parse.ly spelling in tests comments. [#561](https://github.com/Parsely/wp-parsely/pull/561)
- Minor amendments on the documentation. [#514](https://github.com/Parsely/wp-parsely/pull/514)
- Updated release process guidelines. [#567](https://github.com/Parsely/wp-parsely/pull/567)
- Removed checkboxes from GitHub's PR template. [#512](https://github.com/Parsely/wp-parsely/pull/512)
- Improved JS scripts integration tests. [#557](https://github.com/Parsely/wp-parsely/pull/557)
- WordPress.org screenshots for settings page. [#574](https://github.com/Parsely/wp-parsely/pull/574)

### Fixed

- Undefined index error on settings page. [#536](https://github.com/Parsely/wp-parsely/pull/536)
- Source the correct asset for the Recommendations Widget. [#545](https://github.com/Parsely/wp-parsely/pull/545)
- End-to-end tests in CI (GitHub Actions). [#521](https://github.com/Parsely/wp-parsely/pull/521)

> - Source code linting violations (PHPCS with `--serverity=1`). [#544](https://github.com/Parsely/wp-parsely/pull/544)

Reviewer 2: I'd move the phpcs change to the "Changed" section instead of "Fixed."

- Source the correct asset for the Recommendations Widget. [#545](https://github.com/Parsely/wp-parsely/pull/545)
- End-to-end tests in CI (GitHub Actions). [#521](https://github.com/Parsely/wp-parsely/pull/521)

### Removed

- Plugin version number being printed in front-end's HTML source code. [#502](https://github.com/Parsely/wp-parsely/pull/502)
- Custom CSS on wp-admin settings page. [#496](https://github.com/Parsely/wp-parsely/pull/496)
- `migrate_old_fields` private function from Recommended Widget. [#599](https://github.com/Parsely/wp-parsely/pull/599)
- PHP 8.2 from CI tests. [#523](https://github.com/Parsely/wp-parsely/pull/523)
- Custom end-to-end Docker image. [#524] (https://github.com/Parsely/wp-parsely/pull/524)

## [3.0.4] - 2022-01-17

### Changed

- Changed plugin loading functions from anonymous to named functions. [#595](https://github.com/Parsely/wp-parsely/pull/595)

## [3.0.3] - 2022-01-12

### Fixed

- [Fixed a fatal error](https://github.com/Parsely/wp-parsely/issues/587) when requesting metadata for a post without categories and `categories as tags` enabled. [#588](https://github.com/Parsely/wp-parsely/pull/588)

> 

Reviewer: `develop` should have this content in it now they've been tagged, so this PR should be rebased onto `develop` so that it only includes the 3.1.0 changes.
